tests/dialects/test_doris.py

```diff
@@ -18,3 +18,56 @@ def test_regex(self):
                 "doris": "SELECT REGEXP(abc, '%foo%')",
             },
         )
+
+    def test_concat(self):
+        self.validate_all(
+            "SELECT a || b",
+            write={
+                "doris": "SELECT CONCAT(a, b)",
+            },
+        )
+
+    def test_to_date(self):
+        self.validate_all(
+            "SELECT TO_DATE('2019-12-01')",
+            read={
+                "oracle": "SELECT TO_DATE('2019-12-01')",
+                "hive": "SELECT TO_DATE('2019-12-01')",
+            },
+            write={
+                "doris": "SELECT TO_DATE('2019-12-01')",
+            },
+        ),
+        self.validate_all(
+            "SELECT STR_TO_DATE('2019-12-01', 'yyyy-MM-dd')",
+            read={
+                "oracle": "SELECT TO_DATE('2019-12-01', 'yyyy-MM-dd')",
+            },
+            write={
+                "doris": "SELECT STR_TO_DATE('2019-12-01', 'yyyy-MM-dd')",
+            },
+        ),
+        self.validate_all(
+            "SELECT STR_TO_DATE('2019-12-01', '%Y-%m-%d')",
+            write={
+                "doris": "SELECT STR_TO_DATE('2019-12-01', '%Y-%m-%d')",
+            },
+        ),
+        self.validate_all(
+            "NVL2(x, y, z)",
+            write={
+                "doris": "CASE WHEN x IS NOT NULL THEN z ELSE y END",
+            },
+        ),
+        self.validate_all(
+            "to_char('2009-10-04 22:23:00', 'yyyymm')",
+            write={
+                "doris": "DATE_FORMAT('2009-10-04 22:23:00', 'yyyymm')",
+            },
+        ),
```

Comment:
Since `TO_CHAR` and `NVL2` are not supported by Doris, these tests should be of the form:

```python
self.validate_all(
    "CASE WHEN x IS NOT NULL THEN z ELSE y END",
    read={
        "snowflake": "NVL2(x, y, z),
    },
)
```